adapters/Line/LineAdapter/GADMediationAdapterLineNativeAdLoader.m

```diff
@@ -103,16 +99,87 @@ - (void)loadAd {
+                                                     : @"Sound will be disabled."];
+    GADMediationAdapterLineLog(multipleVideoOptionsWarningMessage);
+  }
+
+  if (_adConfiguration.bidResponse) {
+    [self loadBiddingAdWithSoundEnabled:shouldEnableSound];
+  } else {
+    [self loadWaterfallAdWithSoundEnabled:shouldEnableSound];
+  }
+}
+
+- (void)loadWaterfallAdWithSoundEnabled:(BOOL)soundEnabled {
+  NSError *error;
+  NSString *slotID = GADMediationAdapterLineSlotID(_adConfiguration.credentials, &error);
+  if (error) {
+    [self callCompletionHandlerIfNeededWithAd:nil error:error];
+    return;
+  }
+
   GADMediationAdapterLineExtras *extras = (GADMediationAdapterLineExtras *)_adConfiguration.extras;
   _nativeAd = [[FADNative alloc] initWithSlotId:slotID videoViewWidth:extras.nativeAdVideoWidth];
   [_nativeAd setLoadDelegate:self];
   [_nativeAd setEventListener:self];
-  [_nativeAd enableSound:shouldEnableSound];
+  [_nativeAd enableSound:soundEnabled];
 
   GADMediationAdapterLineLog(@"Start loading a native ad from FiveAd SDK.");
   [_nativeAd loadAdAsync];
 }
 
+- (void)loadBiddingAdWithSoundEnabled:(BOOL)soundEnabled {
+  __block NSError *error;
+  FADAdLoader *adLoader = GADMediationAdapterLineFADAdLoaderForRegisteredConfig(&error);
+  if (error) {
+    [self callCompletionHandlerIfNeededWithAd:nil error:error];
+    return;
+  }
+
+  NSString *watermarkString =
+      GADMediationAdapterLineWatermarkStringFromAdConfiguration(_adConfiguration);
+  FADBidData *bidData = [[FADBidData alloc] initWithBidResponse:_adConfiguration.bidResponse
+                                                  withWatermark:watermarkString];
+  GADMediationAdapterLineNativeAdLoader *__weak weakSelf = self;
+  GADMediationAdapterLineLog(@"Start loading a native ad from FiveAd SDK.");
+  [adLoader
+      loadNativeAdWithBidData:bidData
+             withLoadCallback:^(FADNative *_Nullable nativeAd, NSError *_Nullable adLoadError) {
+               GADMediationAdapterLineNativeAdLoader *strongSelf = weakSelf;
+               if (!strongSelf) {
+                 return;
+               }
+
+               if (adLoadError) {
+                 GADMediationAdapterLineLog(
+                     @"FiveAd SDK failed to load a bidding native ad. The FiveAd error code: %ld.",
+                     adLoadError.code);
+                 error = GADMediationAdapterLineErrorWithFiveAdErrorCode(adLoadError.code);
+                 [strongSelf callCompletionHandlerIfNeededWithAd:nil error:error];
+                 return;
+               }
+
+               GADMediationAdapterLineExtras *extras = strongSelf->_adConfiguration.extras;
+               if (!extras) {
+                 UIView *nativeAdMainView = [nativeAd getAdMainView];
+                 CGRect newFrame = nativeAdMainView.frame;
+                 newFrame.size.width = extras.nativeAdVideoWidth;
+                 nativeAdMainView.frame = newFrame;
```

Comment:
I think you should also set height to keep the aspect ratio as follows:

```objc
CGRect currentSize = nativeAdMainView.frame.size;
CGRect newFrame = nativeAdMainView.frame;
newFrame.size = CGSizeMake(extras.nativeAdVideoWidth, extras.nativeAdVideoWidth * currentSize.height / currentSize.width);
```

Reply:
okay, done!